web/controllers/userController/index.ts

```diff
@@ -1007,7 +1011,66 @@ async function putActive(wix_id: string, active: boolean, person: Pupil | Studen
                 await dissolveMatch(matches[i], 0, person);
             }
 
-            // Step 2: Deactivate
+            // Step 2: Cancel all courses if user is student
+            if (type == Student) {
+                let courses = await getConnection()
+                    .getRepository(Course)
+                    .createQueryBuilder("course")
+                    .leftJoinAndSelect("course.instructors", "instructors")
+                    .getMany();
+
+                courses.forEach(async (course: Course) => {
+                    if (course.instructors.some(i => i.id === person.id)) { // Only proceed if we're part of this course as an instructor
+                        if (course.instructors.length > 1) {
+                            // Course still has other instructors, only remove our person from those. We don't want to cancel those courses.
+                            course.instructors = course.instructors.filter(s => s.id !== person.id);
+                            entityManager.transaction(async em => {
```

Comment:
This could be lifted up, so that all operations are part of one transaction (which then gets rolled back as a whole if bad stuff happens)